cmd/kubelet/app/server.go

```diff
@@ -344,13 +456,36 @@ func run(s *options.KubeletServer, kcfg *KubeletConfig) (err error) {
 			}
 		}
 	}
-	if c, err := configz.New("componentconfig"); err == nil {
-		c.Set(s.KubeletConfiguration)
+
+	// Register current configuration with /configz endpoint
+	cfgz, err := configz.New("componentconfig")
+	if err == nil {
+		tmp := kubeExternal.KubeletConfiguration{}
+		api.Scheme.Convert(&s.KubeletConfiguration, &tmp, nil)
+		cfgz.Set(tmp)
 	} else {
 		glog.Errorf("unable to register configz: %s", err)
 	}
 
 	if kcfg == nil {
+		// Look for config on the API server. If it exists, replace s.KubeletConfiguration
+		// with it and continue. initKubeletConfigSync also starts the background thread that checks for new config.
+
```

Comment:
FYI, feature gates for alpha features is in #30468. Can you define a `DynamicKubeletConfig` key [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/config/feature_gate.go#L34) and check the value of `DefaultFeatureGates.DynamicKubeletConfig()` to decide whether to use dynamic config?